pkg/controller/pkg/revision/hook.go

```diff
@@ -105,6 +108,15 @@ func (h *ProviderHooks) Post(ctx context.Context, pkg runtime.Object, pr v1alpha
 		return errors.Wrap(err, errApplyProviderDeployment)
 	}
 	pr.SetControllerReference(runtimev1alpha1.Reference{Name: d.GetName()})
+
+	for _, c := range d.Status.Conditions {
+		if c.Type == appsv1.DeploymentAvailable {
+			if c.Status == v1.ConditionTrue {
+				return nil
+			}
```

Comment:
Do we have an existing helper for this - `resource.IsConditionTrue` iirc?

Reply:
Unfortunately our helper only works for our `controller-runtime` conditions